### PR TITLE
Fix supports_device in DummyBackend

### PR DESCRIPTION
**Pull request by houseroad**
Add the implementation in DummyBackends.

Also turn the default value to False in the base class.

### CODE REVIEW

File: onnx/backend/base.py [hunk 1]
```diff
@@ -75,4 +75,4 @@ def supports_device(cls, device):
         Checks whether the backend is compiled with particular device support.
         In particular it's used in the testing suite.
         """
-        return True
+        return False
```
[the remark this answers is not in the document]
Comment (houseroad) [reply onnx/backend/base.py]: I am open to discuss this choice. Turning to False will be safer. It's just the whitelist or blacklist we want to use.

[the remark this answers is not in the document]
Comment (bddppq) [reply onnx/backend/base.py]: Yes, whitelist and blacklist are equally expressive, but I think the driving factor is whether we want backends to opt-in or opt-out to running tests.

File: onnx/test/test_backend_test.py [hunk 2]
```diff
@@ -35,6 +35,13 @@ def run_node(cls, node, inputs, device='CPU', outputs_info=None):
         raise BackendIsNotSupposedToImplementIt(
             "This is the dummy backend test that doesn't verify the results but does run the checker")
 
+    @classmethod
+    def supports_device(cls, device):
```
[the remark this answers is not in the document]
Comment (houseroad) [reply onnx/test/test_backend_test.py]: In this case, checking each model once should be good enough, I think. (We can reduce the CI time, too)

**Comment by raymondxyang** [comment]
<https://github.com/onnx/onnx/blob/master/onnx/backend/test/runner/__init__.py#L192> CUDA is manually set as a default value.. probably we can pass only CPU into it

**Comment by houseroad** [comment]
@raymondxyang that is also intended. Because we don't really invoke any CUDA related code, it's fine to enable it in ONNX repo. The correct implementation of backend will override supports_device anyway, so no worries.

I think the root cause is OOM due to python inefficient memory management. :-)

**Comment by raymondxyang** [comment]
Got it @houseroad so shall we disable the alexnet case for CI?

**Comment by houseroad** [comment]
@raymondxyang I think no need to disable it. Let's see the Appveyor's result first.

**Comment by bddppq** [comment]
@anderspapitto Could you merge DummyBackend in backend_shape_inference_test.py into test_backend_test.py?

**Pull request merged**